app/src/test/java/com/rickyhu/hushkeyboard/ui/HomeScreenUiTest.kt

```diff
@@ -32,4 +39,46 @@ class HomeScreenUiTest {
             .assertIsEnabled()
             .assertHasClickAction()
     }
+
+    @Test
+    fun `Select input button should exist, should be enable, and should have click action`() {
+        composeTestRule
+            .onNodeWithText("Select input method")
+            .assertExists()
+            .assertIsEnabled()
+            .assertHasClickAction()
+    }
+
+    @Test
+    fun `The textField should be displayed and should be enabled to enter text`() {
+        composeTestRule
+            .onNodeWithText("Type here")
+            .assertIsDisplayed()
+            .performTextInput("Test input")
+
+        composeTestRule.onNodeWithText("Test input").assertIsDisplayed()
+    }
+
+    @Test
+    fun `Settings button should exist, should be enabled, and should have click action`() {
+        composeTestRule
+            .onNodeWithContentDescription("Settings")
+            .assertExists()
+            .assertIsEnabled()
+            .assertIsDisplayed()
+            .assertHasClickAction()
+    }
+
+    @Test
+    fun `Settings button should disappear and action should be invoked upon click`() {
+        composeTestRule
+            .onNodeWithContentDescription("Settings")
+            .performClick()
+
+        composeTestRule
+            .onNodeWithText("Settings")
+            .assertIsNotDisplayed()
+
+        assert(clicked)
```

Comment:
To test this assert, I think I would move `setContent` block into each test case, since this is the only test case that needs to examine if the click action is really performed. Consider removing `setUp` so that you can also move `clicked` property to a local variable in this test case.

Besides, I recommend using `assetTrue()` from the JUnit library instead of using `asset` from Kotlin. Remember to import the usages as well.
```suggestion
        assertTrue(clicked)
```